core/src/main/java/org/mule/runtime/core/processor/TryMessageProcessor.java

```diff
@@ -0,0 +1,159 @@
+ * If the {@link org.mule.runtime.core.api.transaction.TransactionConfig} is null then no transaction is used and the next
+ * {@link org.mule.runtime.core.api.processor.Processor} is invoked directly.
+ */
+public class TryMessageProcessor extends AbstractMessageProcessorOwner implements Processor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TryMessageProcessor.class);
+
+  protected List<Processor> processors = emptyList();
+  protected MessageProcessorChain nestedChain;
+  protected MuleTransactionConfig transactionConfig;
+
+  @Override
+  public Event process(final Event event) throws MuleException {
+    if (nestedChain == null) {
+      return event;
+    } else {
+      ExecutionTemplate<Event> executionTemplate =
+          createScopeExecutionTemplate(muleContext, flowConstruct, transactionConfig, messagingExceptionHandler);
+      ExecutionCallback<Event> processingCallback = () -> nestedChain.process(event);
+
+      try {
+        return executionTemplate.execute(processingCallback);
+      } catch (MuleException e) {
+        throw e;
+      } catch (Exception e) {
+        throw new DefaultMuleException(errorInvokingMessageProcessorWithinTransaction(nestedChain, transactionConfig), e);
+      }
+    }
+  }
+
+  @Override
+  public Publisher<Event> apply(Publisher<Event> publisher) {
+    if (nestedChain == null) {
+      return publisher;
+    } else if (isTransactionActive() || transactionConfig.getAction() != ACTION_INDIFFERENT) {
+      return Processor.super.apply(publisher);
+    } else {
+      return from(publisher).transform(nestedChain);
+    }
+  }
+
+  /**
+   * Configures the {@link MessagingExceptionHandler} that should be used to handle any errors that occur in this scope.
+   *
+   * @param exceptionListener the {@link MessagingExceptionHandler} to be used.
+   */
+  public void setExceptionListener(MessagingExceptionHandler exceptionListener) {
+    this.messagingExceptionHandler = exceptionListener;
+  }
+
+  /**
+   * Configures the {@link TransactionConfig} that that defines the transactional behaviour of this scope.
+   *
+   * @param transactionConfig the {@link TransactionConfig} to be used.
+   */
+  public void setTransactionConfig(MuleTransactionConfig transactionConfig) {
+    this.transactionConfig = transactionConfig;
+  }
+
+  /**
+   * Obtain the {@link TransactionConfig} configured that defines transactional behaviour of this scope.
+   *
+   * @return the configured {@link TransactionConfig}.
+   */
+  public MuleTransactionConfig getTransactionConfig() {
+    return transactionConfig;
+  }
+
+  /**
+   * Configure the nested {@link Processor}'s that error handling and transactional behaviour should be applied to.
+   * 
+   * @param processors
+   */
+  public void setMessageProcessors(List<Processor> processors) {
+    this.processors = processors;
```

Comment:
You should create the chain here, I don't see why you'd need the processors referenced by themselves.